src/Microsoft.AspNet.Mvc.Razor/LanguageViewLocationExpander.cs

```diff
@@ -21,36 +22,32 @@ namespace RazorWebSite
     /// </example>
     public class LanguageViewLocationExpander : IViewLocationExpander
     {
-        private const string ValueKey = "language";
-        private readonly Func<ActionContext, string> _valueFactory;
-
-        /// <summary>
-        /// Initializes a new instance of <see cref="LanguageViewLocationExpander"/>.
-        /// </summary>
-        /// <param name="valueFactory">A factory that provides tbe language to use for expansion.</param>
-        public LanguageViewLocationExpander(Func<ActionContext, string> valueFactory)
-        {
-            _valueFactory = valueFactory;
-        }
-
         /// <inheritdoc />
         public void PopulateValues(ViewLocationExpanderContext context)
         {
-            var value = _valueFactory(context.ActionContext);
-            if (!string.IsNullOrEmpty(value))
-            {
-                context.Values[ValueKey] = value;
-            }
         }
 
         /// <inheritdoc />
         public virtual IEnumerable<string> ExpandViewLocations(ViewLocationExpanderContext context,
                                                                IEnumerable<string> viewLocations)
         {
-            string value;
-            if (context.Values.TryGetValue(ValueKey, out value))
+#if DNX451
+            var cultureInfo = Thread.CurrentThread.CurrentUICulture;
+#else
+            var cultureInfo = CultureInfo.CurrentUICulture;
+#endif
+
+            var updatedViewLocations = new List<string>();
+            while (cultureInfo != cultureInfo.Parent)
+            {
+                updatedViewLocations.AddRange(ExpandViewLocationsCore(viewLocations, cultureInfo.Name));
+                cultureInfo = cultureInfo.Parent;
```

Comment:
From MSDN:
> The cultures have a hierarchy in which the parent of a specific culture is a neutral culture, the parent of a neutral culture is the InvariantCulture, and the parent of the InvariantCulture is the invariant culture itself. The parent culture encompasses only the set of information that is common among its children.

Should we be checking `cultureInfo.IsNeutralCulture`? as our stop condition?